cajita/src/Cajita_ParticleGridDistributor.hpp

```diff
@@ -0,0 +1,390 @@
+                {
+                    if ( periodic[d] )
+                    {
+                        if ( positions( p, d ) > global_high[d] )
+                            positions( p, d ) -= global_extent[d];
+                        else if ( positions( p, d ) < global_low[d] )
+                            positions( p, d ) += global_extent[d];
+                    }
+                }
+            } );
+    }
+
+  private:
+    LocalGridType _local_grid;
+};
+
+//-----------------------------------------------------------------------//
+/*!
+  \brief Check for the number of particles that must be communicated
+
+  \tparam LocalGridType Cajita LocalGrid type.
+
+  \tparam PositionSliceType Particle position type.
+
+  \param local_grid The local grid containing periodicity and system bound
+  information.
+
+  \param positions The particle position container, either Slice or View.
+
+  \param minimum_halo_width Number of halo mesh widths to include for
+  ghosting.
+*/
+template <class LocalGridType, class PositionSliceType>
+int migrateCount( const LocalGridType& local_grid,
+                  const PositionSliceType& positions,
+                  const int minimum_halo_width )
+{
+    ParticleGridDistributor<LocalGridType> pg_distributor( local_grid );
+
+    return pg_distributor.migrateCount( positions, minimum_halo_width );
+}
+
+//---------------------------------------------------------------------------//
+/*!
+  \brief Determine which data should be migrated from one uniquely-owned
+  decomposition to another uniquely-owned decomposition, using bounds of a
+  Cajita grid and taking periodic boundaries into account.
+
+  \tparam LocalGridType Cajita LocalGrid type.
+
+  \tparam PositionSliceType Position type.
+
+  \param local_grid The local grid containing periodicity and system bound
+  information.
+
+  \param positions The particle positions.
+
+  \return Distributor for later migration.
+*/
+template <class LocalGridType, class PositionSliceType>
+Cabana::Distributor<typename PositionSliceType::device_type>
+createParticleGridDistributor( const LocalGridType& local_grid,
+                               PositionSliceType& positions )
+{
+    using device_type = typename PositionSliceType::device_type;
+
+    ParticleGridDistributor<LocalGridType> pg_distributor( local_grid );
+
+    // Get all 26 neighbor ranks.
+    auto topology = pg_distributor.getTopology();
+
+    Kokkos::View<int*, Kokkos::HostSpace, Kokkos::MemoryUnmanaged>
+        topology_host( topology.data(), topology.size() );
+    auto topology_mirror =
+        Kokkos::create_mirror_view_and_copy( device_type(), topology_host );
+    Kokkos::View<int*, device_type> destinations(
+        Kokkos::ViewAllocateWithoutInitializing( "destinations" ),
+        positions.size() );
+
+    // Determine destination ranks for all particles and wrap positions across
+    // periodic boundaries.
+    pg_distributor.getMigrateDestinations( topology_mirror, destinations,
```

Comment:
Same here - could just inject code into free function instead of having a class.

Reply:
I should definitely have commented to make this free function wrapper more clear, but this was again just to use the mesh spatial dimension without the user having to worry about it

Reply:
Removed